k2/torch/bin/decode.cc

```diff
@@ -0,0 +1,89 @@
+#include <cassert>
+#include <cstdio>
```

Comment:
It would be nice if it were  more obvious from the binary itself what it does and how to use it.. or at least a pointer to where to find this info.

Reply:
mm, I see it has a usage message; is it accurate? (./bin/decode.py)... and what does this binary do?
I'm just wondering what kind of thing a jit file can contain, is it a nnet evaulation?  With fixed size?

Reply:
This is just a prototype. I will add more documentation later.

The intention of this file is to replicate what we are doing in `decode.py` in icefall.

Reply:
> I'm just wondering what kind of thing a jit file can contain, is it a nnet evaulation? With fixed size?

As far as I know,  the torch scripted model behaves similarly to `torch.nn.Module`. You can vary the input shape, i.e.,
the `N` and `T` in `(N, T, C)`.

Checkpoints just save the parameters of a model, without topology information.
If you want to recover from some checkpoint, you have to first build a `nn.Module` through some Python code, 
which contains the definition of the model.

With a torch scripted model, all you need is a binary file ( I don't know what exactly is saved in it), load
it in C++, feed some input into it, and get the output for later processing.

Reply:
the jit file has model weights and the model code translated into TorchScript intermediate representation (IR). AFAIK it doesn’t speed up the inference in any way unless the IR code is further optimized; e.g. it can be translated/exported to ONNX. 

Also I think it is possible to extend both TorchScript and ONNX with custom ops (by building shared library extensions: https://pytorch.org/tutorials/advanced/torch_script_custom_ops) — maybe it makes sense to build k2 custom ops and involve LM + decoding as part of the JIT model for deployment packaging, I guess a lot of people would find it very convenient (same could be done for kaldifeat for a single-file model that outputs transcripts directly). Some food for thought.

Reply:
Oh, that’s too bad. I guess it’s possible to convert an FSA to a bunch of tensors and reconstruct it inside but it’s probably too much hassle…